### DISCUSSION TITLE
“Gerechtigkeit für alle” or how to make `roll` applicable for admins

### DISCUSSION THREAD

**Discussion by egregors** [vldc-hq/vldc-bot#255] [General]
It's been a long time, but #93 is still here. Guys, we really should figure out how to fix it.
I believe here could be a few different solutions, how can we restrict admins. 

* [hard way] admin got shot -> Nyan revoke admin role from “dead” admin -> Nyan applying restrictions as for regular hussar -> after resurrection, Nyan is giving admin role to user back
  * ? idk is bot even can revoke or grand admin role for users?
* [easy way] do not revoke admin role, but remove all messages from “dead” admin from the chat (like in `silence_mode`).

Both solutions are kind of controversial. In the first one, admin become unable to perform his admin's responsibilities, until he leaves the club. In the second one, naughty admin can easily `unmute` himself by himself.  

Maybe you guys have another options to solve it. Feel free to share any relevant ideas!

UPD: active RP: #256

**Comment by RWB01** [comment]
I fully support the suggestion to remove roles from admins, restrict them as common users and return roles just right after. It's an honor to be restricted for any hussar! Hussars don't have to be ashamed by non-restrictable shameless admins.

**Comment by korenskoy** [comment]
Предлагаю лишать лычек и банить!

**Comment by egregors** [comment]
So, okay. Das stimmt. Ich werde kompliziert Weg zu probieren. #256 

Ehrlich gesagt, es sieht nicht so schlecht. Aber ich kenne nicht, wie und wann soll man „Admin rolle“ zurückgeben.
